pandas/core/indexes/base.py

```diff
@@ -177,10 +179,15 @@ class Index(IndexOpsMixin, PandasObject):
     str = accessor.AccessorProperty(strings.StringMethods)
 
     def __new__(cls, data=None, dtype=None, copy=False, name=None,
-                fastpath=False, tupleize_cols=True, **kwargs):
+                fastpath=False, tupleize_cols=True, names=None,
+                **kwargs):
 
-        if name is None and hasattr(data, 'name'):
-            name = data.name
+        if name is None:
```

Comment:
actually move all of this names stuff to a separate method

```
def _validate_names(self, name, names):
    .....
```

because we need to call this in subclasses

MultiIndex will override this (to allow names as a list), the basic one will raise if ``name`` is not a scalar

Reply:
I can be wrong, but isn't `_validate_names` used only when we try to create a copy of an object? It doesn't affect initial object creation. And if I add a new method specially for these pre-checks it might be a bit confusing.